source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -569,7 +569,11 @@ namespace
                 if (!RtApi::get().hasMasterPrim(RtToken(nodedef->getName())))
                 {
                     PvtPrim* prim = readNodeDef(nodedef, stage);
-                    RtNodeDef(prim->hnd()).registerMasterPrim();
+                    RtNodeDef rtnodedef(prim->hnd());
+                    if (!rtnodedef.isMasterPrim())
```

Comment:
The if statement above should be covering this, so this should not be needed. Unless there is a bug with nodedef naming.. `nodedef->getName()` should give the same name as `rtnodedef.getName()` here, which is used inside `isMasterPrim()`.

Reply:
Oh, it looks like Jerry committed something for this before me!